src/plugins/source.js

```diff
@@ -1,11 +1,12 @@
 import path from 'path'
 import fs from 'fs'
+import _ from 'lodash'
 
 export default async function () {
   const content = await this.driver.getSource()
 
   return new Promise((resolve) => {
-    const pathToLog = path.resolve(process.cwd(), 'appium_source.xml')
+    const pathToLog = path.resolve(process.cwd(), _.uniqueId('appium_source_')+'.xml')
```

Comment:
I think better add date+time as unique label.
appium_source_2019-07-22_10:24.xml

Reply:
Also, need to return the final source file name because we put this file into artifacts at a CI.

Reply:
Okay, I've made both of the changes you've requested - the date format though is ISO8601 standard YYYYMMDDTHHmmss.  The ':' character is reserved for filenames on some systems.